### PR TITLE
feat: keep last selected agent in storage

### PR DESCRIPTION
**Pull request by mohandast52**
## Proposed changes

https://github.com/user-attachments/assets/8610f748-b296-4c71-b46c-cfbb2277a583

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/components/AgentSelection.tsx [hunk 1]
```diff
@@ -52,7 +52,8 @@ const EachAgent = ({
         // if the selected type is Memeooorr - should set up the agent first
         gotoPage(Pages.Setup);
         gotoSetup(SetupScreen.SetupYourAgent);
-      } else {
+      } else if (agentType === AgentType.PredictTrader) {
+        gotoPage(Pages.Setup);
```
Comment (Tanya-atatakai) [inline frontend/components/AgentSelection.tsx:56]: Thanks, I missed that 🙇‍♀️

File: frontend/service/agents/PredictTrader.ts [hunk 2]
```diff
@@ -182,11 +195,14 @@ export abstract class PredictTraderService extends StakedAgentService {
...
+  ): Promise<StakingContractDetails | undefined> => {
     const { multicallProvider } = PROVIDERS[chainId];
+    const stakingTokenProxy =
+      STAKING_PROGRAMS[chainId][stakingProgramId]?.contract;
 
-    const { contract: stakingTokenProxy } =
-      STAKING_PROGRAMS[chainId][stakingProgramId];
+    if (!PREDICT_TRADER_STAKING_PROGRAM_IDS.includes(stakingProgramId)) {
+      return Promise.resolve(undefined);
+    }
```
Comment (truemiller) [inline frontend/service/agents/PredictTrader.ts:205]: can get check this instead

https://github.com/valory-xyz/olas-operate-app/blob/004c19ba5cc1cf2609d6a456faa351abf3101338/frontend/config/stakingPrograms/gnosis.ts#L28-L44

File: frontend/service/agents/Memeooor.ts [hunk 3]
```diff
@@ -178,11 +184,15 @@ export abstract class MemeooorBaseService extends StakedAgentService {
   static getStakingContractDetails = async (
     stakingProgramId: StakingProgramId,
     chainId: EvmChainId,
-  ): Promise<StakingContractDetails> => {
+  ): Promise<StakingContractDetails | undefined> => {
     const { multicallProvider } = PROVIDERS[chainId];
 
-    const { contract: stakingTokenProxy } =
-      STAKING_PROGRAMS[chainId][stakingProgramId];
+    if (!MEMEOOORR_STAKING_PROGRAMS_IDS.includes(stakingProgramId)) {
```
[the remark this answers is not in the document]
Comment (Tanya-atatakai) [reply frontend/service/agents/Memeooor.ts:190]: Can you explain pls where the mismatch comes from? Can we not rely on hardcoded values based on agent type here, will need to update it every time we add a new agent type

File: frontend/service/agents/PredictTrader.ts [hunk 4]
```diff
@@ -16,6 +16,17 @@ import { ONE_YEAR, StakedAgentService } from './StakedAgentService';
 
 const MECH_REQUESTS_SAFETY_MARGIN = 1;
 
+// TODO: move to common place, enums/StakingProgram.ts?
+const PREDICT_TRADER_STAKING_PROGRAM_IDS = [
```
Comment (Tanya-atatakai) [inline frontend/service/agents/PredictTrader.ts:20]: We have the list of staking programs, it's just based on chain, not agent type. Maybe should rethink this approach, but don't think we should duplicate

File: frontend/service/agents/Memeooor.ts [hunk 5]
```diff
@@ -178,11 +182,11 @@ export abstract class MemeooorBaseService extends StakedAgentService {
   static getStakingContractDetails = async (
     stakingProgramId: StakingProgramId,
     chainId: EvmChainId,
-  ): Promise<StakingContractDetails> => {
+  ): Promise<StakingContractDetails | undefined> => {
     const { multicallProvider } = PROVIDERS[chainId];
 
-    const { contract: stakingTokenProxy } =
-      STAKING_PROGRAMS[chainId][stakingProgramId];
+    const stakingTokenProxy = BASE_STAKING_PROGRAMS[stakingProgramId]?.contract;
```
Comment (Tanya-atatakai) [inline frontend/service/agents/Memeooor.ts:188]: hmm if we know this is base chain, why to pass chainId then? can't we just do smth like:
```
if (!Object.keys(STAKING_PROGRAMS[chainId]).includes(stakingProgramId)) return
```

**Review by Tanya-atatakai — APPROVED** [review]

**Pull request merged**